rdrf/rdrf/views/patient_view.py

```diff
@@ -539,22 +547,44 @@ def get_forms(self, request, registry_model, user, instance=None):
 
         return forms
 
-    def _section_hidden(self, user, registry, fieldlist):
+    def _exclude_hidden_fields(self, user, registry, fieldlist):
+        from rdrf.models.definition.models import DemographicFields
+        user_groups = [g.name for g in user.groups.all()]
+        hidden_fields = DemographicFields.objects.filter(
+            registry=registry, groups__name__in=user_groups, hidden=True, is_section=False
+        ).values_list('field', flat=True)
+        return list(set(fieldlist) - set(hidden_fields))
+
+    def _section_fields_hidden(self, user, registry, fieldlist):
         from rdrf.models.definition.models import DemographicFields
         user_groups = [g.name for g in user.groups.all()]
         hidden_fields = DemographicFields.objects.filter(field__in=fieldlist,
                                                          registry=registry,
-                                                         group__name__in=user_groups,
+                                                         groups__name__in=user_groups,
                                                          hidden=True)
 
         return len(fieldlist) == hidden_fields.count()
 
+    def _section_hidden(self, user, registry, section_name):
+        from rdrf.models.definition.models import DemographicFields
+        user_groups = [g.name for g in user.groups.all()]
+        return DemographicFields.objects.filter(
+            field=f"{DemographicFields.SECTION_PREFIX}{section_name}",
+            registry=registry,
+            groups__name__in=user_groups,
+            is_section=True,
+            hidden=True
+        ).exists()
+
     def _check_for_hidden_section(self, user, registry, form_sections):
         section_hiddenlist = []
```

Comment:
All this nesting `initialisation`, `looping`, `continue`, `appending` isn't just a `filter` operation in disguise? :-)

Reply:
Updated.